src/citrine/_session.py

```diff
@@ -263,15 +263,22 @@ def delete_resource(self, path: str, **kwargs) -> dict:
     @staticmethod
     def _extract_response_json(path, response) -> dict:
         """Extract json from the response or log and return an empty dict if extraction fails."""
+        extracted_response = {}
         try:
-            return response.json()
+            if "application/json" in response.headers.get("Content-Type", ""):
+                extracted_response = response.json()
+            else:
+                # TODO: We are currently skipping this coverage in pytest, mocks need to be made
+                extracted_response = response  # pragma: no cover
```

Comment:
I'm still a little concerned about the fact that these two conditions return different types.